app/tests/integration/nimbus/test_desktop_client_integrations.py

```diff
@@ -190,3 +199,115 @@ def test_check_telemetry_enrollment_unenrollment(
+    create_desktop_experiment,
+    experiment_default_data,
+    check_ping_for_experiment,
+    telemetry_event_check,
+):
+    _row_locator = (By.CSS_SELECTOR, "tr > td > span > span")
+    _search_bar_locator = (By.ID, "about-config-search")
+    wait = WebDriverWait(selenium, 60)
+
+    def wait_function(selenium, wait_string):
+        """This function refreshes about:config waiting for the pref to flip"""
+
+        def _wait_function(selenium=selenium, wait_string=wait_string):
+            try:
+                selenium.get("about:config")
+                search_bar = wait.until(
+                    EC.presence_of_element_located(_search_bar_locator)
+                )
+                search_bar.send_keys("nimbus.qa.pref-1")
+                wait.until(EC.presence_of_element_located(_row_locator))
+                elements = selenium.find_elements(*_row_locator)
+                assert wait_string in [element.text for element in elements]
+            except Exception:
+                time.sleep(2)
+                return False
+            else:
+                return True
+
+        return _wait_function
+
+    requests.delete("http://ping-server:5000/pings")
+    targeting = helpers.load_targeting_configs()[0]
+    experiment_slug = str(slugify(experiment_name))
+    experiment_default_data["targetingConfigSlug"] = targeting
+    experiment_default_data["featureConfigId"] = 9
+    experiment_default_data["referenceBranch"] = {
+        "description": "reference branch",
+        "name": "Branch 1",
+        "ratio": 100,
+        "featureEnabled": True,
+        "featureValue": '{"value": "test_string_automation"}',
+    }
+    experiment_default_data["treatmentBranches"] = [
+        {
+            "description": "treatment branch",
+            "name": "Branch 2",
+            "ratio": 0,
+            "featureEnabled": False,
+            "featureValue": "",
+        }
+    ]
+    create_desktop_experiment(
+        experiment_slug,
+        "desktop",
+        targeting,
+        experiment_default_data,
+    )
+
+    wait.until(wait_function(selenium, "default"))
+
+    summary = SummaryPage(selenium, urljoin(base_url, experiment_slug)).open()
+    summary.launch_and_approve()
+
+    kinto_client.approve()
+
+    summary = SummaryPage(selenium, urljoin(base_url, experiment_slug)).open()
+    summary.wait_for_live_status()
+
+    # Ping the server twice as it sleeps sometimes
+    requests.get("http://ping-server:5000/pings")
+    time.sleep(5)
+
+    # Check there was a telemetry event for the enrollment
+    control = False
+    timeout = time.time() + 60 * 5
+    while control is not True:
+        control = telemetry_event_check(experiment_slug, "enroll")
+        if time.time() > timeout:
+            assert False, "Experiment enrollment was never seen in ping Data"
+    # check experiment exists, this means it is enrolled
+    assert check_ping_for_experiment(experiment_slug), "Experiment not found in telemetry"
```

Comment:
I think it will always return True 🤔

Reply:
It wont return until the experiment is found in the ping. I guess this needs a timeout but if it enrolls, it should always return eventually

Reply:
yeah adding the timeout will make complete sense, thank you for explaining that 😊